### ISSUE TITLE
Add support for UI installers when mods have options.

### ISSUE THREAD

**Issue by nullfrctl** [E1337Kat/cyberpunk2077_ext_redux#347]
**Is your feature request related to a problem? Please describe.**
Always when I'm downloading mods with options, often I will have to manually correct the file structure.
It's nothing difficult, but is definitely tedious and is hard to do when managing several option-filled mods.

It can also mess up the entire structure, such as extra files (item codes, preview pictures, etc.) are not put into extra files folders.

All of this can of course be done manually, but is extremely tedious!

**Describe the solution you'd like**
A good solution, in my opinion, is to convert these types of mods to a UI, such as using a primitive or automatically-generated FOMOD that takes the file structure of the mod in context, like MO2's BAIN installer for Bethesda games.

**Describe alternatives you've considered**
An alternative is to simply maintain the current behavior, which is having to manually adjust mods in the staging folder.

*this is one of my first issues on github, please tell me how to improve them*

**Comment by E1337Kat** [comment]
yes, this is something we would like to have done for sure. We just need to come up with a way to handle it. Perhaps pulling ideas from the FOMOD installer in vortex might be a good idea.

**Comment by nullfrctl** [comment]
Only thing I'd think of is to put into a temporary folder and automatically generate an XML file using folder structure and install from there.

**Comment by effs** [comment]
Related to #344
